_studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw_vaapi.cpp

```diff
@@ -570,7 +568,21 @@ mfxStatus SetFrameRate(
         misc_param->type = VAEncMiscParameterTypeFrameRate;
         frameRate_param = (VAEncMiscParameterFrameRate *)misc_param->data;
 
-        frameRate_param->framerate = TL_attached ? base_framerate / extTL.Layer[par.m_numLayers - 1 - tl].FrameRateScale : base_framerate;
+        if (tl < numTL - 1)
+        {
+            mfxU16 l2lRatio = extTL.Layer[tl + 1].FrameRateScale / extTL.Layer[tl].FrameRateScale;
+            if (num % l2lRatio == 0)
+            {
+                num /= l2lRatio;
+            }
+            else
+            {
+                denom *= l2lRatio;
+            }
```

Comment:
PackMfxFrameRate(...) involves fraction reducing, so no need in optimization here.
denom *= extTL.Layer[tl + 1].FrameRateScale; nom *= extTL.Layer[tl].FrameRateScale;
will work correctly regardless of FrameRateScale ratio.

Reply:
Fixed.